### PR TITLE
Implement Update Actions To Start And Finish

### LINKED ISSUE #3726: Implement updating two chunks of lookup data
This requires to implement the update actions `start` and `finish` to update the lookup data.

### PR DESCRIPTION
**Pull request by mariaschett**
Fixes #3726 .

### CODE REVIEW

File: oak_functions/lookup/src/lib.rs [hunk 1]
```diff
@@ -183,19 +181,50 @@ where
...
-    pub fn update_data(&self, action: UpdateAction, new_data: Data) -> UpdateStatus {
+    pub fn update_start_and_finish(&self, new_data: Data) -> UpdateStatus {
         let mut data_builder = self.data_builder.lock();
+        match &data_builder.state {
+            BuilderState::Empty => {
+                data_builder.extend(new_data);
+                let next_data = data_builder.build();
+                let mut data = self.data.lock();
+                *data = Arc::new(next_data);
+                UpdateStatus::Finished
```
Comment (tiziano88) [inline oak_functions/lookup/src/lib.rs:192]: Maybe worth a comment that this method needs to do the same things as update_start and update_finish (and must be manually kept in sync if they change)

File: oak_functions/lookup/src/lib.rs [hunk 2]
```diff
@@ -281,35 +310,99 @@ mod tests {
...
+
+    // Create test data with size distinct keys between inclusive start and exclusive end.
+    fn create_test_data(start: i32, end: i32) -> Data {
+        let mut map = HashMap::new();
+        for i in start..end {
+            map.insert(
+                format!("key{}", i).into_bytes(),
+                format!("value{}", i).into_bytes(),
+            );
+        }
```
Comment (tiziano88) [inline oak_functions/lookup/src/lib.rs:405]: could probably be something more functional, like `(start..end).map(|i| (format!("key{}", i).into_bytes(), format!("value{}", i).into_bytes())).collect()`

File: oak_functions_service/src/lib.rs [hunk 3]
```diff
@@ -163,35 +163,24 @@ impl schema::OakFunctions for OakFunctionsService {
...
             // UpdateActions.
             _ => Err(micro_rpc::Status::new_with_message(
                 micro_rpc::StatusCode::InvalidArgument,
-                format!("action not implemented: {:?}", self),
+                format!("action not implemented: {:?}", action),
             )),
-        }
+        }?;
+
+        let update_status2 = schema::UpdateStatus::from(update_status);
```
Comment (tiziano88) [inline oak_functions_service/src/lib.rs:180]: The `from` here seems unnecessary given the `into` below, but I'm not 100% sure.

File: oak_functions/lookup/src/lib.rs (hunk 1 above)
Comment (mariaschett) [reply oak_functions/lookup/src/lib.rs:192]: Fixed by calling start and finish with empty data (thanks for the comment :)

File: oak_functions/lookup/src/lib.rs (hunk 2 above)
Comment (mariaschett) [reply oak_functions/lookup/src/lib.rs]: Done.

File: oak_functions_service/src/lib.rs (hunk 3 above)
Comment (mariaschett) [reply oak_functions_service/src/lib.rs]: Curiously `into` did not work , but as i32 did.

**Review by mariaschett — COMMENTED** [review]
Thank you!

**Pull request merged**